### DISCUSSION TITLE
Documenting `automerge` + `lockFileMaintenance` in Configuration Options

### DISCUSSION THREAD

**Discussion by jamesbraza** [renovatebot/renovate#31587] [Suggest an Idea]
### Tell us more.

It looks like from https://github.com/renovatebot/renovate/pull/712, `lockFileMaintenance` now supports automerge. However, looking at [its current docs](https://docs.renovatebot.com/configuration-options/#lockfilemaintenance), this `automerge` capability is not mentioned anywhere.

Can we add `automerge` mentions to the `lockFileMaintenance` somewhere?

**Comment by jamesbraza** [comment]
Same thing goes for `pre-commit`, I am wondering if we can have:

```json5
{
  "pre-commit": {
    enabled: true,
    automerge: true,
  }
}
```

Fwiw this seems to pass with https://github.com/renovatebot/pre-commit-hooks

**Comment by rarkins** [comment]
Automerge works on any PR and is documented independently

**Reply by jamesbraza** [reply]
I guess I don't follow what you meant by "works on any PR".

So let's say I want to:

- Automerge `pre-commit` changes
- Not automerge lock file changes

So I don't think a root level `automerge` can work. Would this be the way to set it up?

```json5
{
  $schema: "https://docs.renovatebot.com/renovate-schema.json",
  extends: ["config:recommended"],
  lockFileMaintenance: {
    enabled: true,
  },
  "pre-commit": {
    enabled: true,
    automerge: true,
  },
}
```

To me, it's unclear the list of sub-key allowable in `lockFileMaintenance` and `pre-commit`. In this case, it's unclear if `automerge` is a valid sub-key of `pre-commit`

**Reply by rarkins** [reply]
Keep it simple and configure automerge using packageRules

**Reply by jamesbraza** [reply]
Please forgive me for re-asking questions, but it's not clear how to configure [`lockFileMaintenance`](https://docs.renovatebot.com/configuration-options/#lockfilemaintenance) with `packageRules`. The `default` for `lockFileMaintenance` doesn't mention `packageRules` at all.

Can you confirm that it would look something like this:

```json
{
  "packageRules": [
    {
      "matchUpdateTypes": ["lockFileMaintenance"],
      "automerge": true,
    }
  ]
}
```

**Reply by jamesbraza** [reply]
I found this: https://docs.renovatebot.com/key-concepts/automerge/#automerge-lock-file-maintenance

It documents the answer to my question. It would be good to link this [here in `lockFileMaintenance`](https://docs.renovatebot.com/configuration-options/#lockfilemaintenance)
